Droas590 iP

Droas590 opened the pull request:
# ScoobyDoo

> “If you only do what you can do, you will never be more than you are now.” ― Master Shifu ([source](https://www.magicalquote.com/character/shifu/))

ScoobyDoo let you do more than you can do, It's
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/Droas590/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :grinning:

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Droas590 committed:
Changed file name, added greetings

Droas590 committed:
Added echo

Droas590 committed:
Added echo

Droas590 committed:
Added list

Droas590 committed:
remove unused import statements

Droas590 committed:
add mark and unmark features

Droas590 committed:
added Deadline Event and Todo

Droas590 committed:
UI testing

Droas590 committed:
add error handling

Droas590 committed:
add delete feature

Droas590 committed:
allow save in local file

Droas590 committed:
Merge branch 'branch-Level-7'
"merge level 7 into master"

Droas590 committed:
added datetime with different format for user

Droas590 committed:
Merge branch 'branch-Level-8'
"merch level 8 to master"

Droas590 committed:
abstract out some part of the program

Droas590 committed:
Merge branch 'branch-A-MoreOOP'
"merge branch more oop into master"

Droas590 committed:
package all the files

Droas590 committed:
Merge branch 'branch-A-Packages'
"merge branch-A-package into master"

Droas590 committed:
Merge branch 'branch-A-gradle'
"add gradle into master"

Droas590 committed:
add tests into parser and events

Droas590 committed:
Merge branch 'branch-A-JUnit'
"merge branch-A-JUnit into master"

Droas590 committed:
add javadocs for all the classes

Droas590 committed:
tweak code to comply with coding standard

Droas590 committed:
add find feature

Droas590 committed:
Merge branch 'branch-A-JavaDoc'
"merge javadoc branch to master"

Droas590 committed:
merge coding standard to master

Droas590 committed:
merge branch-Level-9 to master

Kitty-001 commented on src/main/java/ScoobyDoo/Parser.java:
```diff
@@ -0,0 +1,115 @@
+        if (splitDeadline.length != 2) {
+            throw new InputFormatException("Oops! I need a description to save your task");
+        }
+        String[] splitBySlash = splitDeadline[1].split("/by",2);
```
Perhaps a space after comma?
```suggestion
        String[] splitBySlash = splitDeadline[1].split("/by", 2);
```

Kitty-001 commented on src/main/java/ScoobyDoo/Parser.java:
```diff
@@ -0,0 +1,115 @@
+     */
+    public static LocalDateTime getDeadlineDate(String input) throws InputFormatException {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        String[] splitBy = input.split("/by",2);
```
space after comma~
```suggestion
        String[] splitBy = input.split("/by", 2);
```

Kitty-001 commented on src/main/java/ScoobyDoo/UI.java:
```diff
@@ -0,0 +1,49 @@
+package ScoobyDoo;
```
maybe all lower case for package naming?

charlenetcy replied:
Should Ui.java be in its own Ui package as well?

Kitty-001 reviewed (commented):
LGTM.
Just some coding style details to fix. Well done!!!!!

charlenetcy commented on src/main/java/ScoobyDoo/Parser.java:
```diff
@@ -0,0 +1,115 @@
+package ScoobyDoo;
```
Should Parser.java be in its own package? ie. Parser package

charlenetcy commented on src/main/java/ScoobyDoo/ScoobyDoo.java:
```diff
@@ -0,0 +1,162 @@
+        Scanner scanIn = new Scanner(System.in);
+        while (true) {
+            input = scanIn.nextLine();
+            if (input.equals("bye")) {
```
Should these case checks be handled in Parser.java using switch case to keep ScoobyDoo.java less logic involved?

charlenetcy commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,156 @@
+package storage;
```
Maybe add a line after the package statement
```suggestion
package storage;

```

charlenetcy commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,156 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.stream.Collectors;
+import task.Task;
+import task.Todo;
+import task.Deadline;
+import task.Event;
```
Maybe put these before the other imports and after the package statement
```suggestion
package storage;

import task.Task;
import task.Todo;
import task.Deadline;
import task.Event;
```

charlenetcy commented on src/main/java/storage/Storage.java:
```diff
@@ -0,0 +1,156 @@
+package storage;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.stream.Collectors;
```
Should they be grouped by their packages for easy reference?

charlenetcy reviewed (commented):
Overall, looks good to me. 
I like the name ScoobyDoo for the bot, complements the ToDo behaviour.
I noted a few minor coding standard violations
too.

Droas590 committed:
add gui to ScoobyDoo

Droas590 committed:
add data file to .gitignore

Droas590 committed:
Merge branch 'branch-Level-10'
"merge Level-10-branch to master"

Droas590 committed:
refactor command and parser

abstract out the command from ScoobyDoo.java
into new class.
Each command class represent one command
user can input.

Let's abstract more for better readability.

Droas590 committed:
remove unused code

Droas590 committed:
Use stream for storage and taskList find()

Using streams makes code more readable and understandable

Droas590 committed:
remove unnecessary code

Droas590 committed:
add assertions to ScoobyDoo.java and Storage.java

assertions added to make the code more robust

Droas590 committed:
Merge pull request #2 from Droas590/branch-A-Assertions

add assertions to ScoobyDoo.java and Storage.java

Droas590 committed:
merge master to codequality branch

Droas590 committed:
manually add code from master

Somehow the merging cannot identify the same ScoobyDoo.java file.
So updated code from master is manully added.

Droas590 committed:
Merge branch 'master' into branch-A-Streams
"merge master to branch-A-Stream"

Droas590 committed:
Merge pull request #3 from Droas590/branch-A-CodeQuality

Branch a code quality

Droas590 committed:
change command from delete to remove

Droas590 committed:
delete data file

Droas590 committed:
Merge branch 'master' into branch-A-Streams
"merge master to branch-A-Streams"

Droas590 committed:
Merge pull request #4 from Droas590/branch-A-Streams

Branch a streams

Droas590 committed:
add undo and redo feature

some commands inherited undoable. Two stacks where used to store undo and redo commands for convenient.

Droas590 committed:
remove redundant codes

Droas590 committed:
remove redundant codes

Droas590 committed:
Merge branch 'branch-B-Undo'
"merge branch-B-Undo to master"

Droas590 committed:
alter error message

Droas590 committed:
make user guide

Droas590 committed:
add Ui.png

Droas590 committed:
add user guide

Droas590 committed:
fix bug in parsing
